### PR TITLE
Smooth Transfer

### PR DESCRIPTION
**Pull request by TownEater**
Hello @fblah 

I ran into an issue using this addon where transferring shape keys from a lower poly mesh to a higher one would result in a "stair-step" effect on the destination mesh. I noticed this problem was due to the shape key transfer being based on the change in position of the nearest vertex.

I added a new transfer mode (with a toggle to turn it on and off) called "Smooth Transfer" which instead determines the change in position based on the change of the nearest point on the source surface. This works great for my use case, so I cleaned it up a little and created this pull request to make it available to others. I also incremented the version to 1.0.3 with the change.

Feel free to incorporate this feature if you like it, or ignore it if you don't. Thanks for making the addon in the first place, it's very helpful!

### CODE REVIEW

File: shapekeytransfer.py [hunk 1]
```diff
@@ -109,34 +112,70 @@ def set_vertex_position(self, v_pos):
     # update 1 vertex of destination mesh
     def update_vertex(self):
         if(self.current_vertex_index >= self.total_vertices ):
-            return False
+                return False
```
Comment (fblah) [inline shapekeytransfer.py]: can you fix this indent

**Comment by fblah** [comment]
Hi! Thank you for helping out to add this feature. Could you tell me if you have tested this sufficiently with a few meshes and if its good to be merged. I am a bit busy and will not be able to test this sufficiently.

**Comment by TownEater** [comment]
I'm not sure if it will work with every mesh, there could be edge cases that would cause it to break, but that's a big reason I added this as a new option that can be disabled. If the old algorithm works better for any mesh, it can easily be used instead. Here is what the tools menu looks like with the new smooth transfer option (enabled by default):
Screenshot from 2025-11-15 17-40-00

With all that said, here's some examples to show the difference:

A plane with 4 quads transferring a shape key to a plane with many quads.
Nearest vertex (current algorithm):
Screenshot from 2025-11-15 17-26-48

Smooth Transfer (new algorithm):
Screenshot from 2025-11-15 17-27-13

A blender monkey transferring a shape key to a blender monkey that has had a subdivision surface applied:
Base mesh with shapekey applied:
Screenshot from 2025-11-15 17-31-48

Nearest vertex (current algorithm):
Screenshot from 2025-11-15 17-31-54

Smooth Transfer (new algorithm):
Screenshot from 2025-11-15 17-32-10

Let me know your thoughts and if you'd want me to change anything, or if there's any other mesh examples you'd like to see.

**Comment by fblah** [comment]
Alright the feature definitely offers something more, so I'm going to merge it now. If there are any bugs that pop up later will fix it as it comes!
